src/molecules/IconLink/IconLink.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ButtonHTMLAttributes, DetailedHTMLProps } from 'react';
+import { StyledComponentClass } from 'styled-components';
+import Icon, { IconName } from '../../atoms/Icon';
+import Typography from '../../atoms/Typography';
+import styled from '../../styled-components';
+import Theme from '../../Theme';
+// import { IconName } from '@fortawesome/fontawesome-svg-core';
+
+IconLink.defaultProps = { as: 'a' };
+
+export const IconTypography = styled(Typography)`
+  background: none;
+  border: none;
+  padding: 0;
+  color: ${props => props.theme.text};
+
+  :hover {
+    color: ${props => props.theme.primaryDark};
+    /* stylelint-disable-next-line max-nesting-depth */
+  }
+` as StyledComponentClass<
+  DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
+  Theme
+>;
+
+export function IconLink({
+  href,
+  target,
+  icon,
+  ariaLabel,
+  handleClick,
+}: {
+  href?: string;
+  target?: string;
+  icon: IconName;
+  ariaLabel: string;
+  handleClick?(): void;
+}) {
+  return href ? (
+    <a href={href} target={target} aria-label={ariaLabel}>
+      <IconTypography onClick={handleClick}>
+        <Icon icon={icon} />
+      </IconTypography>
+    </a>
+  ) : (
+    <IconTypography onClick={handleClick} aria-label={ariaLabel}>
+      <Icon icon={icon} />
+    </IconTypography>
+  );
+}
+
+IconTypography.defaultProps = { as: 'button', type: 'button' };
```

Comment:
Please move the defaultProps for IconTypography right after that component

Reply:
Good call